tests/test_scalars.py

```diff
@@ -339,147 +381,219 @@ def test_timedelta() -> None:
+        assert_type(pd.TimedeltaIndex([1, 2, 3], unit="D") < td, np_ndarray_bool),
         np.ndarray,
     )
+    check(assert_type(dt.timedelta(days=1) < td, bool), bool)
+    check(assert_type(ndarray_td64 < td, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td < pd.TimedeltaIndex([1, 2, 3], unit="D"), np.ndarray), np.ndarray
+        assert_type(pd.TimedeltaIndex([1, 2, 3], unit="D") < td, np_ndarray_bool),
+        np.ndarray,
     )
 
     check(assert_type(td > td, bool), bool)
     check(assert_type(td > dt.timedelta(days=1), bool), bool)
     check(assert_type(td > np.timedelta64(1, "D"), bool), bool)
+    check(assert_type(td > ndarray_td64, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td > np.array([1, 2, 3], dtype="timedelta64[D]"), np.ndarray),
+        assert_type(td > pd.TimedeltaIndex([1, 2, 3], unit="D"), np_ndarray_bool),
         np.ndarray,
     )
+    check(assert_type(dt.timedelta(days=1) > td, bool), bool)
+    check(assert_type(ndarray_td64 > td, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td > pd.TimedeltaIndex([1, 2, 3], unit="D"), np.ndarray), np.ndarray
+        assert_type(pd.TimedeltaIndex([1, 2, 3], unit="D") > td, np_ndarray_bool),
+        np.ndarray,
     )
 
     check(assert_type(td <= td, bool), bool)
     check(assert_type(td <= dt.timedelta(days=1), bool), bool)
     check(assert_type(td <= np.timedelta64(1, "D"), bool), bool)
+    check(assert_type(td <= ndarray_td64, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td <= np.array([1, 2, 3], dtype="timedelta64[D]"), np.ndarray),
+        assert_type(td <= pd.TimedeltaIndex([1, 2, 3], unit="D"), np_ndarray_bool),
         np.ndarray,
     )
+    check(assert_type(dt.timedelta(days=1) <= td, bool), bool)
+    check(assert_type(ndarray_td64 <= td, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td <= pd.TimedeltaIndex([1, 2, 3], unit="D"), np.ndarray),
+        assert_type(pd.TimedeltaIndex([1, 2, 3], unit="D") <= td, np_ndarray_bool),
         np.ndarray,
     )
 
     check(assert_type(td >= td, bool), bool)
     check(assert_type(td >= dt.timedelta(days=1), bool), bool)
     check(assert_type(td >= np.timedelta64(1, "D"), bool), bool)
+    check(assert_type(td >= ndarray_td64, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td >= np.array([1, 2, 3], dtype="timedelta64[D]"), np.ndarray),
+        assert_type(td >= pd.TimedeltaIndex([1, 2, 3], unit="D"), np_ndarray_bool),
         np.ndarray,
     )
+    check(assert_type(dt.timedelta(days=1) >= td, bool), bool)
+    check(assert_type(ndarray_td64 >= td, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td >= pd.TimedeltaIndex([1, 2, 3], unit="D"), np.ndarray),
+        assert_type(pd.TimedeltaIndex([1, 2, 3], unit="D") >= td, np_ndarray_bool),
         np.ndarray,
     )
 
     check(assert_type(td == td, bool), bool)
```

Comment:
for any of the `==` tests that return `bool` and `!=` tests that return `bool`, use `assert` and `assert not` to check the result.

Reply:
I decided to use a pattern of pair testing and then an xor, e.g., 

```
a = 1

eq = a == 2
ne = a != 2
assert eq != ne
```

Which will ensure one is `True` and the other `False` without worrying about pandas code correctness.